runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -151,7 +156,11 @@ public static int attachXWalkCore(Context context) {
         Log.d(TAG, "Attach xwalk core");
         sProvisionalInstance = new XWalkCoreWrapper(context, -1);
         if (!sProvisionalInstance.findEmbeddedCore()) {
-            sProvisionalInstance.findSharedCore();
+            if (XWalkUpdater.isDownloadModeEnabled()) {
```

Comment:
The second startup of the app won't trigger the XWalkUpdater. So this way of checking download mode is incomplete